### PR TITLE
[dagit] Replace moment-timezone

### PR DESCRIPTION
**Pull request by hellendag**
### Summary & Motivation

https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore

We no longer need moment-timezone, because we can get the features we need via native the `Intl` API (almost everything) or via the much smaller dayjs (stuff like `fromNow()`).

Replace it throughout Dagit. I'll add comments inline where applicable.

In a followup, I'll remove `moment` entirely.

Cloud needs a TS declaration similar to the ones added here for `Intl`.

### How I Tested These Changes

TS, lint, jest.

- Verify that the Run timeline shows correct dates and times for my timezone.
- View User Settings and open the timezone selector. Verify that the items render correctly (and are sorted and bucketed appropriately) and that I can search and select correctly. Select a timezone (UTC), then view the Run timeline to verify that my new timezone is what I see.
- View Deployment page, verify that the "fromNow" strings for code locations and daemons are correct.
- View time strings throughout Dagit, verify that they look correct.
- View Schedules, verify that human cron strings look correct. Verify that the "latest tick" time and tooltip are correct for individual schedules.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11197** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11197" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11197?utm_source=stack-comment).

File: js_modules/dagit/packages/core/src/instance/DaemonList.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const DaemonList: React.FC<Props> = ({daemonStatuses, showTimestampColumn
   }
 
   return (
-    <Table>
+    <Table $monospaceFont={false}>
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/instance/DaemonList.tsx:49]: Annoyed me that the daemons table was still monospace. Slowly replacing it...

File: js_modules/dagit/packages/core/src/partitions/SliceSlider.tsx [hunk 2]
```diff
@@ -1,60 +0,0 @@
-// eslint-disable-next-line no-restricted-imports
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/partitions/SliceSlider.tsx:1]: This is no longer used!

File: js_modules/dagit/packages/core/src/ui/TimeFromNow.tsx [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 
-export const TimeFromNow: React.FC<{unixTimestamp: number}> = ({unixTimestamp}) => {
+export const TimeFromNow: React.FC<{unixTimestamp: number}> = React.memo(({unixTimestamp}) => {
   return (
     <Tooltip
       placement="top"
       content={<Timestamp timestamp={{unix: unixTimestamp}} timeFormat={TIME_FORMAT} />}
     >
-      {moment.unix(unixTimestamp).fromNow()}
+      {dayjs(unixTimestamp * 1000).fromNow()}
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/ui/TimeFromNow.tsx:17]: Only use case of dayjs so far.

**Comment by hellendag** [comment]
Ran some build comparisons. The main bundle file drops from 630KB to 579KB.

Before:

Screenshot 2022-12-19 at 1 46 23 PM

After:

Screenshot 2022-12-19 at 1 43 36 PM

File: js_modules/dagit/packages/core/typings.d.ts [hunk 4]
```diff
@@ -28,3 +28,13 @@ declare module 'worker-loader!*' {
...
 
 declare module 'chartjs-adapter-date-fns';
+
+declare namespace Intl {
+  type Key = 'calendar' | 'collation' | 'currency' | 'numberingSystem' | 'timeZone' | 'unit';
+
+  function supportedValuesOf(input: Key): string[];
+
+  interface Locale extends LocaleOptions {
+    timeZones: string[];
```
Comment (bengotow) [inline js_modules/dagit/packages/core/typings.d.ts:38]: I wonder if we could get this into our env by changing something in our base CRA fork? Doesn't seem super important though and this is likely easier, just imagine having to update it a bit as we continue to use more of Intl

File: js_modules/dagit/packages/core/src/ui/TimeFromNow.tsx (hunk 3 above)
Comment (bengotow) [reply js_modules/dagit/packages/core/src/ui/TimeFromNow.tsx:17]: Nice! Glad we did a good job of getting this relative time + tooltip everywhere.

File: js_modules/dagit/packages/core/src/runs/RunTimeline.tsx [hunk 5]
```diff
@@ -328,6 +327,16 @@ const TimeDividers = (props: TimeDividersProps) => {
     });
   }, [locale, timeZone]);
 
+  const dateFormatWithTimezone = React.useMemo(() => {
+    return Intl.DateTimeFormat(locale, {
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/runs/RunTimeline.tsx:331]: Good call memoizing this 👍

**Review by bengotow — APPROVED** [review]
This looks great! I knew moment-timezone was big but it's wild this saves us that much space...

File: js_modules/dagit/packages/core/typings.d.ts (hunk 4 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/typings.d.ts:38]: My bet is that it will be included in future TS versions.

**Pull request merged**